webapp/src/ts/services/android-api.service.ts

```diff
@@ -27,7 +34,22 @@ export class AndroidApiService {
     private simprintsService:SimprintsService,
     private zone:NgZone,
     private routeSnapshotService:RouteSnapshotService,
+    private headerTabsService:HeaderTabsService,
   ) {
+    this.subscribeToStore();
+    this.getPrimaryTab();
+  }
+
+  private subscribeToStore() {
+    this.store
+      .select(Selectors.getCurrentTab)
+      .subscribe(currentTab => this.currentTab = currentTab);
+  }
+
+  private getPrimaryTab() {
+    this.headerTabsService
+      .getFirstAuthorizedTab()
+      .then(tab => this.primaryTab = tab);
```

Comment:
It turns out that android's evaluate javascript function doesn't like promises... I had to store the primary tab so back() function is sync.